File: 1-two-sum/two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        index_map = {}
        for idx, num in enumerate(nums):
            complement = target - num
            if complement in index_map:
                return [index_map[complement], idx]
            index_map[num] = idx

File: 1-two-sum/two-sum.go
func twoSum(nums []int, target int) []int {
    store := make(map[int]int)

    for i, x := range nums {
        store_i, exist := store[x]
        if exist {
            return []int{store_i, i}
        } else {
            store[target - x] = i
        }
    }

    return []int{}
}